### PR TITLE
Kernelize Lift, Errors, Context

### PR DESCRIPTION
**Pull request by Odomontois**

### CODE REVIEW

File: concurrent/src/main/scala/tofu/concurrent/ContextT.scala [hunk 1]
```diff
@@ -71,7 +71,7 @@ object ContextT extends ContextTInstances {
 
 }
 
-trait ContextTInstances extends ContextTInstancesQ
+trait ContextTInstances extends ContextTInstancesP
```
Comment (FunFunFine) [inline concurrent/src/main/scala/tofu/concurrent/ContextT.scala:74]: I think it's better to use meanigful name, like ContextTInstancesPrimary or something else.
Source code readability matters here, reader should be able to understand why some instances are in another trait.

Scaladoc is another option

File: core/src/test/scala/tofu/ContextSuite.scala [hunk 2]
```diff
@@ -14,9 +14,9 @@ object ContextSuite {
     implicitly[WithContext[ReaderT[F, Ctx, *], Ctx]]
     implicitly[HasLocal[ReaderT[F, Ctx, *], Ctx]]
     implicitly[WithLocal[ReaderT[F, Ctx, *], Ctx]]
-    implicitly[HasProvide[ReaderT[F, Ctx, *], F, Ctx]]
+    implicitly[HasProvide[ReaderT[F, Ctx, *], F, Ctx]](ContextBase.readerTContext[F, Ctx])
```
Comment (catostrophe) [inline core/src/test/scala/tofu/ContextSuite.scala:17]: is explicit instance still needed even after aliasing HasProvide to WithProvide?

File: core/src/test/scala/tofu/lift/UnliftSuite.scala [hunk 3]
```diff
@@ -50,6 +51,7 @@ object UnliftSuite {
   }
 
   def summonUnliftIOInstances1[F[_]: Effect, R](): Unit = {
+    // Unlift.unliftIOEffect[IO]
```
Comment (catostrophe) [inline core/src/test/scala/tofu/lift/UnliftSuite.scala:54]: ?

File: concurrent/src/main/scala/tofu/concurrent/impl/contextt.scala [hunk 4]
```diff
@@ -319,8 +319,7 @@ final class ContextTRunContext[F[+_]: Applicative, C[_[_]]](implicit FD: Defer[F
 
 // instance that does not defer locals. could be stack-unsafe
 final class ContextTRunContextUnsafe[F[+_]: Applicative, C[_[_]]]
-    extends ContextTContext[F, C] with RunContext[ContextT[F, C, *]]                       {
-  type Lower[+A] = F[A]
+    extends ContextTContext[F, C] with WithRun[ContextT[F, C, *], F, C[ContextT[F, C, *]]] {
```
Comment (FunFunFine) [inline concurrent/src/main/scala/tofu/concurrent/impl/contextt.scala:322]: Why exactly do we need it?

**Review by FunFunFine — APPROVED** [review]

File: core/src/test/scala/tofu/ContextSuite.scala (hunk 2 above)
Comment (Odomontois) [reply core/src/test/scala/tofu/ContextSuite.scala:17]: oh, this is garbage, sorry

File: core/src/test/scala/tofu/lift/UnliftSuite.scala (hunk 3 above)
Comment (Odomontois) [reply core/src/test/scala/tofu/lift/UnliftSuite.scala:54]: -[] remove garbage in UnliftSuite

File: concurrent/src/main/scala/tofu/concurrent/impl/contextt.scala (hunk 4 above)
Comment (Odomontois) [reply concurrent/src/main/scala/tofu/concurrent/impl/contextt.scala:322]: To drop Defer requirement when it's supposed to be fine

File: kernel/src/main/scala/tofu/lift/Unlift.scala [hunk 5]
```diff
@@ -24,14 +24,14 @@ object Lift extends LiftInstances1 {
       def lift[A](fa: F[A]): G[A] = fk(fa)
     }
 
-  private val liftIdentityAny: Lift[AnyK, AnyK] = new Lift[AnyK, AnyK] {
+  private val liftIdentityAny: Lift[Any, Any] = new Lift[Any, Any] {
```
[the remark this answers is not in the document]
Comment (Odomontois) [reply kernel/src/main/scala/tofu/lift/Unlift.scala:27]: - [ ] reintroduce AnyK

File: concurrent/src/main/scala/tofu/concurrent/ContextT.scala (hunk 1 above)
Comment (Odomontois) [reply concurrent/src/main/scala/tofu/concurrent/ContextT.scala:74]: It's just Z, Y, X, .... here, just like 1, 2, 3... in other instance ladders

Comment (Odomontois) [reply concurrent/src/main/scala/tofu/concurrent/ContextT.scala:74]: We can change this after kernelization

File: concurrent/src/main/scala/tofu/concurrent/impl/contextt.scala [hunk 6]
```diff
@@ -7,7 +7,7 @@ import cats.syntax.bifunctor._
 import cats.tagless.InvariantK
 import tofu.syntax.funk.funKFrom
 import tofu.syntax.monadic._
-import tofu.{RunContext, WithContext, WithRun}
+import tofu.{WithContext, WithRun}
```
Comment (FunFunFine) [inline concurrent/src/main/scala/tofu/concurrent/impl/contextt.scala:10]: Filename isn't right

**Review by FunFunFine — APPROVED** [review]

File: concurrent/src/main/scala/tofu/concurrent/impl/contextt.scala (hunk 6 above)
Comment (Odomontois) [reply concurrent/src/main/scala/tofu/concurrent/impl/contextt.scala:10]: Why?

**Review by catostrophe — DISMISSED** [review]

**Review by catostrophe — APPROVED** [review]

**Pull request merged**